logbook-core/src/main/java/org/zalando/logbook/Conditions.java

```diff
@@ -70,18 +71,18 @@
 
     public static Predicate<RawHttpRequest> header(final String key, final String value) {
         return request ->
-                request.getHeaders().containsEntry(key, value);
+                request.getHeaders().getOrDefault(key, Collections.emptyList()).contains(value);
     }
 
     public static Predicate<RawHttpRequest> header(final String key, final Predicate<String> predicate) {
         return request ->
                 request.getHeaders().get(key).stream().anyMatch(predicate);
```

Comment:
This will throw an exception now if a key doesn't exist. Can you verify with a test?